src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchValidateFsm.java

```diff
@@ -269,6 +282,32 @@ protected void expectedEntitiesBuild(SwitchValidateState from, SwitchValidateSta
         fireReadyIfAllResourcesReceived();
     }
 
+    protected void errorWhileCollectingData(SwitchValidateState from, SwitchValidateState to,
+                                            SwitchValidateEvent event, SwitchValidateContext context) {
+        log.info("Caught error while collecting data, checking source (switch={}, key={}", getSwitchId(), key);
+        MessageCookie sourceCookie = context.getRequestCookie();
+        if (sourceCookie == null) {
+            return;
+        }
+        CommandData request = requestsTable.remove(sourceCookie.getValue());
+        if (request == null) {
+            log.warn("Request got by cookie is null, aborting, (cookie={})", sourceCookie);
+            return;
```

Comment:
I think we should not exit from this method without firing some event. Because in this case we will just stuck in state `COLLECT_DATA` if error message was received at the end. I think we should fire `ERROR` before these two `returns`. And add log message before first `return`

Reply:
+1

Reply:
Agree with you